build-logic/build.gradle.kts

```diff
@@ -38,11 +38,18 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
  * SOFTWARE.
  */
-package org.graalvm.buildtools;
 
-// Do not edit this file directly!
-// Edit 'version.properties' in root of the repository and then use 'common/scripts/updateVersions.sh' to keep everything in sync.
-public class VersionInfo {
-    public static final String JUNIT_PLATFORM_NATIVE_VERSION = "0.9.2-SNAPSHOT";
+plugins {
+    `kotlin-dsl`
+    id("com.github.joschi.licenser") version "0.6.1"
```

Comment:
Yes, it verifies license headers _and_ updates them. The same can be done with checkstyle + IDE, yes, although I have found that my IDE doesn't consistently create license headers. It's also not granted that anyone would use an IDE to develop. That said:

- the checkstyle rules only verifies existing source sets (not build files, not samples, ...)
- this plugin is a build time dependency, not a runtime one, with AL2 in any case. It's not even used for _user code_, I can barely imagine how this could be a problem.

Reply:
Thing is that there is no optimal solution to this issue. 

Checkstyle currently serves as a guard against merging unlicensed (or wrong licensed files) as it forces the user to update headers manually. Yes this is tedious, however automatic tool might create issues like [this](https://github.com/graalvm/native-build-tools/pull/87/files/b6ca784221ca9c183e8f02e8a592eb1cefa30df9#r663263153) which could possibly fall through (due to all of the tests passing). I believe that build files and samples could be included in the checkstyle definition as well (for the build files there is [file filter](https://github.com/melix/native-build-tools/blob/051df2f64bf0b0a29f012ba032a85d518a763fb3/config/checkstyle.xml#L244) change as well as forcing checkstyle to include files which are not in the sourcesets).

As you said correctly, user could work without IDE, but if I'm not mistaken, user can also not run the check task and push unlicensed code to the PR. At the end this is a trade-off between developer convenience and reviewer convenience, where in the main GraalVM repository we focus on later (since developer is responsible for making sure that his code is properly formatted).

As Graeme said that build time dependencies are OK, I'm fine with merging this change, even though I would personally avoid this path.